docs/root/version_history/current.rst

```diff
@@ -45,6 +45,7 @@ New Features
 * access loggers: extened specifier for FilterStateFormatter to output :ref:`unstructured log string <config_access_log_format_filter_state>`.
 * access loggers: file access logger config added :ref:`log_format <envoy_v3_api_field_extensions.access_loggers.file.v3.FileAccessLog.log_format>`.
 * aggregate cluster: make route :ref:`retry_priority <envoy_v3_api_field_config.route.v3.RetryPolicy.retry_priority>` predicates work with :ref:`this cluster type <envoy_v3_api_msg_extensions.clusters.aggregate.v3.ClusterConfig>`.
+* api: a warning is now logged when v2 xDS api is used. This behavior can be temporarily disabled by setting `envoy.reloadable_features.enable_deprecated_v2_api_warning` to `false`.
```

Comment:
This is not a new feature. Please put in the right section. Probably the one above this one.

/wait

Reply:
Hmm, that doesn't sound like a "Removed Config or Runtime" to me. Doesn't really fit any other section either, I don't think?

Reply:
cc @ggreenway @alyssawilk  where do you think this should go? I think that category makes the most sense to me based on what this is about but we might need to rejigger slightly (the categories are new).

Reply:
Hmm, I think it's either a new feature or a bug fix, depending on if we think it's a bug that this didn't used to produce a warning.

Reply:
Could also be put in the deprecation section at the bottom given it's part of the v2 deprecation cycle.  I think if we'd done it with deprecation it would go there, but given it was off a bit it could also go in bugfixes.

I'd be mildly inclined to have it in the deprecation section just because it'll be in the correct place for folks who pull releases but I'm good with whatever folks decide.

Reply:
> I'd be mildly inclined to have it in the deprecation section just because it'll be in the correct place for folks who pull releases but I'm good with whatever folks decide.

+1 that SGTM. @ggreenway wdyt?

Reply:
SGTM.

OTOH, it almost doesn't need a release note for this behavior exactly, but more of a notice in the deprecation section that v2 is going away in X more releases (3?).